pkg/templatelookup/moduletemplateinfolookup/by_version_strategy_test.go

```diff
@@ -69,6 +69,87 @@ func Test_ByVersion_Strategy_Lookup_ReturnsModuleTemplateInfo(t *testing.T) {
 	assert.Equal(t, moduleTemplate.Spec.Channel, moduleTemplateInfo.ModuleTemplate.Spec.Channel)
 }
 
+func Test_ByVersion_Strategy_Lookup_WhenMoreThanOneModuleTemplateFound(t *testing.T) {
```

Comment:
This one should also be possible easily with a client stub returning an error:
![image](https://github.com/user-attachments/assets/706864dd-c72f-41df-bacb-a7ceef010f39)